torch/optim/swa_utils.py

```diff
@@ -47,7 +47,10 @@ def swa_update(averaged_param_list: PARAM_LIST, current_param_list: PARAM_LIST,
             torch._foreach_lerp_(averaged_param_list, current_param_list, 1 / (num_averaged + 1))
         else:
             diffs = torch._foreach_sub(current_param_list, averaged_param_list)
-            torch._foreach_addcdiv_(averaged_param_list, diffs, [num_averaged + 1] * len(averaged_param_list))
+            if isinstance(num_averaged, Tensor):
+                torch._foreach_addcdiv_(averaged_param_list, diffs, [num_averaged + 1] * len(averaged_param_list))
+            else:
+                torch._foreach_add_(averaged_param_list, diffs, alpha=1 / (num_averaged + 1))
```

Comment:
```suggestion
                torch._foreach_add_(averaged_param_list, diffs, alpha=1.0 / (num_averaged + 1))
```
if num_averaged is an int?

Reply:
Thanks for fixing this.